### ISSUE TITLE
Should we be using socket.io with elm?

### ISSUE THREAD

**Issue by finnhodgkin** [dwyl/hapi-socketio-redis-chat-example#93]
If the eventual aim is to move to Pheonix sockets, would it make more sense to skip socket.io and go for plain websockets in our Elm prototype? The only elm-socketio package I've found is no longer maintained (recommends Socket.io 1.4.4) so we'd have to use ports more extensively. 

A potential pitfall is that elm-lang/websocket [doesn't support](https://github.com/elm-lang/websocket/issues/5) long polling so no ie8/9. 

In contrast Elm-Pheonix-socket:

>By default, supports both websockets and longpoll transports.

So in the long run we'll be golden :+1: It also has a similar _ish_ api to elm-lang websockets.

**Comment by nelsonic** [comment]
@finnhodgkin are you referring to this one: https://github.com/mgold/elm-socketio ? 🤔 
I _agree_ with the idea of moving away from Socket.io for the Phoenix-backed app.
Can you please link to where you used _native_ sockets in your Hapi app?

**Comment by finnhodgkin** [comment]
Yep that's the one. My search wasn't very extensive so there may be a nicer, more up to date package. Today's the day for a less speculative look :+1:

Here's our hapi/ws/elm prototype from a few days ago: https://github.com/finnhodgkin/Elm-Online-Offline/tree/elm-heroku-test

**Comment by nelsonic** [comment]
@finnhodgkin your prototype looks good. 👍 
https://github.com/finnhodgkin/Elm-Online-Offline/blob/b36c7b51fd0a98d650b30f74da9f046908604f0b/node-sockets/server.js
I'm _happy_ for you to sub out Socket.io for Native WebSockets, but please write a couple of tests before you dive in. 😉

**Comment by ZooeyMiller** [comment]
We've ended up going with socket.io, so that we can have the Elm chat talk to the non-elm chat and vice versa, and so that we can learn how to work with ports a little more.

Having dealt with Elm's own WebSocket library in the online/offline app that Finn linked we felt learning how to do it with ports would be beneficial, and mean that it fits better in this repo.

If we took out socket.io we'd more or less have to build an entire new back end which doesn't make much sense when the focus for this is front end with Elm. 

If @finnhodgkin is satisfied with this I'll close this issuen
